testing/upload_validation.ts

```diff
@@ -227,6 +228,120 @@ export function validateSyncTableSchema(
   );
 }
 
+/**
+ * Returns a map of sync table names to their child sync table names, or undefined if the hierarchy is invalid.
+ * Example valid return: { Parent: 'Child' }
+ * {} is also a valid result, when there are no sync tables, or no parent relationships.
+ * @hidden
+ */
+export function validateCrawlHierarchy(
+  pack: PackVersionDefinition,
+  context?: z.RefinementCtx,
+): Record<string, string[]> | undefined {
+  const result: Record<string, string[]> = {};
+  if (!pack.syncTables) {
+    return result;
+  }
+  const syncTableSchemasByName: Record<string, ObjectSchema<any, any>> = {};
+  for (const syncTable of pack.syncTables) {
+    syncTableSchemasByName[syncTable.name] = syncTable.schema;
+  }
+  for (const [index, syncTable] of pack.syncTables.entries()) {
+    let parent: string | undefined;
+    for (const param of syncTable.getter.parameters) {
+      if (!param.crawlStrategy) {
+        continue;
+      }
+      if (param.crawlStrategy.parentTable) {
+        const {tableName, propertyKey} = param.crawlStrategy.parentTable;
```

Comment:
Might be worth calling this local variable `parentTableName` for clarity?

Reply:
Very smart.